### PR TITLE
Datepicker Modal Dialog: Fixed next/prev month bug and improve adherence with APG code style guide

### PR DESCRIPTION
**Pull request by jongund**
This PR complements #2618 to provide the remaining changes to resolve issue #2581.

This PR changes behavior of commands for next and previous month and year In the date picker dialog when the currently focused date does not exist in the previous or next month or year. With these changes, if the same day number does not exist in the newly displayed month, the last day of the newly displayed month receives focus.

In addition to the above enhancement, this PR also includes the following changes:
* Updates coding practices to use `class` constructor and pointer events
* Updates documentation of page up and page down commands
* Updates regression tests for page up and page down commands

[Preview updated Date Picker Modal Dialog Example](https://deploy-preview-208--aria-practices.netlify.app/aria/apg/patterns/dialog-modal/examples/datepicker-dialog/)

### Review checklist

* [x] [Editorial review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#editorial_review) by @mcking65
* [x] [Functional review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#functional_review) by @mcking65
* [x] [Visual design review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#design_review) by @ariellalgilmore 
* [x] [Code review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#code_review) by @ariellalgilmore 
* [x] [Test review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#test_review) by @jugglinmike & @paul-clue

___
[WAI Preview Link](https://deploy-preview-208--aria-practices.netlify.app/ARIA/apg) _(Last built on Tue, 16 May 2023 19:24:24 GMT)._

### CODE REVIEW

**Review by ariellalgilmore — APPROVED** [review]

**Review by mcking65 — APPROVED** [review]
Functional and editorial review complete. Good to go.

**Comment by a11ydoer** [comment]
@alflennik would it be possible for you to review Datepicker Modal Dialog update so that we can merge this?

**Comment by ccanash** [comment]
Hi @a11ydoer , Alex had an issue running the test locally and will need additional time to do this review so we will have to put it into our triaging processes. I will provide an update next week.

File: test/tests/dialog-modal_datepicker.js [hunk 1]
```diff
@@ -801,34 +838,30 @@ ariaTest(
...
-    let day = new Date();
+    const startDate = '1/31/2023';
+    const targetDates = [
+      '1/31/2022',
+      '1/31/2021',
+      '1/31/2020',
+      '1/31/2019',
+      '1/31/2018',
+      '1/31/2017',
+    ];
```
Comment (jugglinmike) [inline test/tests/dialog-modal_datepicker.js:849]: This doesn't demonstrate the bug fix (it passes on the `main` branch). Starting from 2020-02-29 would do it, though

```suggestion
    const startDate = '2/29/2020';
    const targetDates = [
      '2/28/2019',
      '2/28/2018',
      '2/28/2017',
      '2/29/2016',
      '2/28/2015',
      '2/28/2014',
    ];
```

**Review by jugglinmike — CHANGES_REQUESTED** [review]

File: test/tests/dialog-modal_datepicker.js [hunk 2]
```diff
@@ -45,6 +45,41 @@ const ex = {
...
+    },
+    ex.inputSelector,
+    m,
+    d,
+    y
+  );
+};
+
+const checkDate = async function assertAriaRoles(t, dateSelector, day) {
+  const d = day.getDate() < 10 ? '0' + day.getDate() : day.getDate();
```
[the remark this answers is not in the document]
Comment (jongund) [reply test/tests/dialog-modal_datepicker.js]: @jugglinmike 
Thank you for the suggestion, I updated the code.

File: test/tests/dialog-modal_datepicker.js (hunk 1 above)
Comment (jongund) [reply test/tests/dialog-modal_datepicker.js]: @jugglinmike 
Thank you for the feedback, I updated the code.

File: test/tests/dialog-modal_datepicker.js [hunk 3]
```diff
@@ -45,6 +45,41 @@ const ex = {
...
+      document.querySelector(inputSelector).value = `${month}/${day}/${year}`;
+    },
+    ex.inputSelector,
+    m,
+    d,
+    y
+  );
+};
+
+const checkDate = async function assertAriaRoles(t, dateSelector, day) {
```
[the remark this answers is not in the document]
Comment (jongund) [reply test/tests/dialog-modal_datepicker.js]: @jugglinmike 
Thank you for the feedback, I updated the code.

**Comment by jongund** [comment]
@jugglinmike @mcking65 
Ready to review updated test cases

**Review by jugglinmike — APPROVED** [review]
The tests look good to @paul-clue and me, @jongund! Thanks!

**Review by mcking65 — APPROVED** [review]
Thank you @jongund for all your work on this!! This is now ready to ship!

**Pull request merged**